File: leetcode/validSudoku.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public boolean isValidSudoku(char[][] board) {
        Set<String> row = new HashSet<>();
        Set<String> col = new HashSet<>();
        Set<String> grid = new HashSet<>();

        for (int idx = 0; idx < 81; idx++) {
            int i = idx / 9;
            int j = idx % 9;

            if (board[i][j] == '.') {
                continue;
            }

            if (row.contains("" + i + board[i][j])
                    || col.contains("" + j + board[i][j])
                    || grid.contains("" + (i / 3) + (j / 3) + board[i][j])) {
                return false;
            }
            row.add("" + i + board[i][j]);
            col.add("" + j + board[i][j]);
            grid.add("" + (i / 3) + (j / 3) + board[i][j]);
        }
        return true;
    }
}

File: leetcode/validsudoku.js
function isValidSudoku(board) {
  const row = new Set();
  const col = new Set();
  const grid = new Set();

  for (let index = 0; index < 81; index++) {
    const i = Math.floor(index / 9);
    const j = index % 9;

    if (board[i][j] === '.') {
      continue;
    }

    if (
      grid.has(`${Math.floor(i / 3)}${Math.floor(j / 3)}${board[i][j]}`)
      || row.has(`${i}${board[i][j]}`)
      || col.has(`${j}${board[i][j]}`)
    ) {
      return false;
    }

    row.add(`${i}${board[i][j]}`);
    col.add(`${j}${board[i][j]}`);
    grid.add(`${Math.floor(i / 3)}${Math.floor(j / 3)}${board[i][j]}`);
  }

  return true;
}

board = [
    ["8","3",".",".","7",".",".",".","."]
,["6",".",".","1","9","5",".",".","."]
,[".","9","8",".",".",".",".","6","."]
,["8",".",".",".","6",".",".",".","3"]
,["4",".",".","8",".","3",".",".","1"]
,["7",".",".",".","2",".",".",".","6"]
,[".","6",".",".",".",".","2","8","."]
,[".",".",".","4","1","9",".",".","5"]
,[".",".",".",".","8",".",".","7","9"]]

console.log(isValidSudoku(board))